### PR TITLE
Add feature flag and enable controllers (GKE part 5)

### PR DESCRIPTION
**Pull request by richardcase**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This change adds the GKE feature flag to CAPG and adds the controllers to main.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relates #512 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Enables GKE support via new feature flag.
```

### CODE REVIEW

File: cloud/scope/cluster.go [hunk 1]
```diff
@@ -54,7 +51,7 @@ func NewClusterScope(ctx context.Context, params ClusterScopeParams) (*ClusterSc
 	}
 
 	if params.GCPServices.Compute == nil {
-		computeSvc, err := createComputeService(ctx, params)
+		computeSvc, err := createComputeService(ctx, params.GCPCluster.Spec.CredentialsRef, params.Client)
```
Comment (afzalbin64) [inline cloud/scope/cluster.go:54]: Caught my eye while exploring CAPG... IMO, we can add `ctx` init like `ctx := context.Background()` before passing it. 😄

**Comment by richardcase** [comment]
As the GKE functionality isn't being used we can ignore any apidiff errors:

/override pull-cluster-api-provider-gcp-apidiff

File: exp/controllers/gcpmanagedcluster_controller.go [hunk 2]
```diff
@@ -212,15 +210,15 @@ func (r *GCPManagedClusterReconciler) reconcile(ctx context.Context, clusterScop
 	record.Event(clusterScope.GCPManagedCluster, "GCPManagedClusterReconcile", "Ready")
 
 	controlPlaneEndpoint := clusterScope.GCPManagedControlPlane.Spec.Endpoint
+	clusterScope.SetControlPlaneEndpoint(controlPlaneEndpoint)
+
 	if controlPlaneEndpoint.IsZero() {
 		log.Info("GCPManagedControlplane does not have endpoint yet. Reconciling")
 		record.Event(clusterScope.GCPManagedCluster, "GCPManagedClusterReconcile", "Waiting for control-plane endpoint")
-		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```
Comment (richardchen331) [inline exp/controllers/gcpmanagedcluster_controller.go:218]: do we still want to requeue in this case?

File: cloud/scope/managedmachinepool.go [hunk 3]
```diff
@@ -67,28 +66,26 @@ func NewManagedMachinePoolScope(ctx context.Context, params ManagedMachinePoolSc
...
+	// var credentialData []byte
+	// var err error
+	// if params.GCPManagedCluster.Spec.CredentialsRef != nil {
+	// 	credentialData, err = getCredentialDataFromRef(ctx, params.GCPManagedCluster.Spec.CredentialsRef, params.Client)
+	// } else {
+	// 	credentialData, err = getCredentialDataFromMount()
+	// }
+	// if err != nil {
+	// 	return nil, errors.Errorf("failed to get credential data: %v", err)
+	// }
```
Comment (richardchen331) [inline cloud/scope/managedmachinepool.go:78]: these should be deleted?

**Comment by richardchen331** [comment]
Thanks for adding the GKE feature flag!

File: exp/controllers/gcpmanagedcluster_controller.go (hunk 2 above)
Comment (richardcase) [reply exp/controllers/gcpmanagedcluster_controller.go:218]: No we don't as there is nothing left for the controller to do until the control plane has updated its endpoint....and we have a watch for that.

File: cloud/scope/managedmachinepool.go (hunk 3 above)
Comment (richardcase) [reply cloud/scope/managedmachinepool.go:78]: Ah yes, missed that...thanks :)

**Review by cpanato — APPROVED** [review]
nice, thanks
/approve

**Pull request merged**